community/modules/scripts/spack-install/templates/install_spack.tpl

```diff
@@ -155,8 +153,13 @@ echo "$PREFIX Populating defined buildcaches"
 %{endfor ~}
 
 if [ ! -f /etc/profile.d/spack.sh ]; then
-        echo "source ${INSTALL_DIR}/share/spack/setup-env.sh" > /etc/profile.d/spack.sh
-        chmod a+rx /etc/profile.d/spack.sh
+        cat <<EOF > /etc/profile.d/spack.sh
```

Comment:
I am not used to seeing text after the <<EOF and it looks odd to me. How does this render. Does the `> /etc/profile.d/spack.sh` end up at the end? It must for this to work. If it does then close this comment.

Reply:
This cats the block from until EOF into the file `/etc/profile.d/spack.sh`.

Reply:
`<< .. >` is just combined input heredoc-style redirection with output redirection you're probably familiar with

I would use `<<-` but it requires the use of tabs and I don't want to mix whitespace styles in this file.